hack/lib/init.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# This script is meant to be the entrypoint for OpenShift Bash scripts to import all of the support
+# libraries at once in order to make Bash script preambles as minimal as possible. This script recur-
+# sively `source`s *.sh files in this directory tree as long as none of the functions declared in the
+# file have been imported yet.
+
+if [[ -z "${OS_ROOT:-}" ]]; then
+	echo "[ERROR] In order to import OpenShift Bash libraries, \$OS_ROOT must be set."
+	exit 1
+fi
+
+library_files=( $( find "${OS_ROOT}/hack/lib" -type f -name '*.sh' -not -path '*/hack/lib/init.sh' ) )
+# TODO(skuzmets): Move the contents of the following files into respective library files.
+library_files+=( "${OS_ROOT}/hack/common.sh" )
+library_files+=( "${OS_ROOT}/hack/util.sh" )
+library_files+=( "${OS_ROOT}/hack/cmd_util.sh" )
+
+for library_file in "${library_files[@]}"; do
+	# we assume that everyone writing library functions is following the style guidelines and 
+	# writing their function definitions like:
+	# function os::function_name() {
+	#     ...
+	# }
+	library_defined_functions=()
+	IFS=$'\n' library_function_definitions=( $( grep -E "^function .*\(\) \{$" "${library_file}" ) )
```

Comment:
@Miciah I considered breaking off a shell, source-ing the file, and doing a diff on the declared functions before and after in order to get a more stable list of functions declared in the file, but I determined that, while this approach is more fragile, we don't lose out on too much by using it. I'd appreciate your thoughts.